gui/wxpython/datacatalog/catalog.py

```diff
@@ -88,6 +90,12 @@ def showDataStructureInfo(self):
     def showImportDataInfo(self):
         self.infoManager.ShowImportDataInfo(self.OnImportOgrLayers, self.OnImportGdalLayers)
 
+    def showImportSuccessfulInfo(self):
+        if self.show_info_after_import:
+            wx.CallAfter(self.parent.SetSelectionByName, 'catalog')
```

Comment:
This is a bad dependency making data catalog dependent on lmgr. The dependency is and should be the other way around. lmgr depends on data catalog while `self.parent.SetSelectionByName, 'catalog'` makes assumption that the parent has a `SetSelectionByName` function which sets focus/shows things and that data catalog is called `catalog`.

How to achieve this is to have signal showMe in data catalog. lmgr can connect to it and do what is needed. This will make data catalog work smoothly in e.g. g.gui.datacatalog where this signal will simple not be connected to anything.

If CallAfter is really needed, the `emit` should be wrapped in it, i.e., it should stay here. It is the data catalog which needs to finish things first, so lmgr should not have the responsibility to do things right.

Reply:
I get it. I have created signal showTab, and I no longer use CallAfter - seems useless.